### PR TITLE
Add a Vala test case for mixed sources

### PR DESCRIPTION
**Pull request by arteymix**
It is numbered to follow #634.

### CODE REVIEW

File: test cases/vala/11 mixed sources/meson.build [hunk 1]
```diff
@@ -0,0 +1,5 @@
+project('foo', 'c', 'vala')
+
+glib = dependency('glib-2.0')
```
Comment (ignatenkobrain) [inline test cases/vala/11 mixed sources/meson.build]: actually I think once we specify 'vala' in langs, it should be automatically and passed over all vala sources

Comment (arteymix) [inline test cases/vala/11 mixed sources/meson.build]: You mean implicit `c`? I don't think Jussi agreed much when it was suggested.

**Comment by arteymix** [comment]
I wonder now how easily the opposite could be done.

**Comment by jpakkane** [comment]
Dependencies may come from external sources or subprojects. We should not hardcode which one Vala uses.

**Comment by arteymix** [comment]
Also, GLib might become optional in the future and I think you can remove that with ``--nostdpkg``.

**Comment by nirbheek** [comment]
glib is optional, but C isn't

**Comment by arteymix** [comment]
@nirbheek until someone decides to write a LLVM backend..

**Pull request merged**